common/flatpak-dir.c

```diff
@@ -2278,169 +2276,174 @@ flatpak_dir_repo_lock (FlatpakDir   *self,
 }
 
 const char *
-flatpak_deploy_data_get_appdata_version (GVariant *deploy_data)
+flatpak_deploy_data_get_appdata_version (GBytes *deploy_data)
 {
   return flatpak_deploy_data_get_string (deploy_data, "appdata-version");
 }
 
 const char *
-flatpak_deploy_data_get_appdata_license (GVariant *deploy_data)
+flatpak_deploy_data_get_appdata_license (GBytes *deploy_data)
 {
   return flatpak_deploy_data_get_string (deploy_data, "appdata-license");
 }
 
 const char *
-flatpak_deploy_data_get_appdata_content_rating_type (GVariant *deploy_data)
+flatpak_deploy_data_get_appdata_content_rating_type (GBytes *deploy_data)
 {
-  g_autoptr(GVariant) metadata = g_variant_get_child_value (deploy_data, 4);
-  const char *value = NULL;
+  VarDeployDataRef ref = var_deploy_data_from_bytes (deploy_data);
+  VarMetadataRef metadata = var_deploy_data_get_metadata (ref);
+  VarVariantRef rating_v;
 
-  g_variant_lookup (metadata, "appdata-content-rating", "(&s*)", &value, NULL);
+  if (var_metadata_lookup (metadata, "appdata-content-rating", NULL,  &rating_v))
+    {
+      VarContentRatingRef rating = var_content_rating_from_variant (rating_v);
+      return var_content_rating_get_rating_type (rating);
+    }
 
-  return value;
+  return NULL;
 }
 
 GHashTable *  /* (transfer container) (nullable) */
-flatpak_deploy_data_get_appdata_content_rating (GVariant *deploy_data)
+flatpak_deploy_data_get_appdata_content_rating (GBytes *deploy_data)
 {
-  g_autoptr(GVariant) metadata = g_variant_get_child_value (deploy_data, 4);
-  const char *id = NULL, *val = NULL;
+  VarDeployDataRef ref = var_deploy_data_from_bytes (deploy_data);
+  VarMetadataRef metadata = var_deploy_data_get_metadata (ref);
+  VarVariantRef rating_v;
   g_autoptr(GHashTable) content_rating = g_hash_table_new_full (g_str_hash, g_str_equal, NULL, NULL);
-  g_autoptr(GVariantIter) iter = NULL;
 
-  if (!g_variant_lookup (metadata, "appdata-content-rating", "(*a{ss})", NULL, &iter))
-    return NULL;
+  if (var_metadata_lookup (metadata, "appdata-content-rating", NULL,  &rating_v))
+    {
+      VarContentRatingRef rating = var_content_rating_from_variant (rating_v);
+      VarRatingsRef ratings = var_content_rating_get_ratings (rating);
+      gsize len, i;
+
+      len = var_ratings_get_length (ratings);
+      for (i = 0; i < len; i++)
+        {
+          VarRatingsEntryRef entry = var_ratings_get_at (ratings, i);
 
-  while (g_variant_iter_loop (iter, "{&s&s}", &id, &val))
-    g_hash_table_insert (content_rating,
-                         (gpointer) g_intern_string (id),
-                         (gpointer) g_intern_string (val));
+          g_hash_table_insert (content_rating,
+                               (gpointer) g_intern_string (var_ratings_entry_get_key (entry)),
+                               (gpointer) g_intern_string (var_ratings_entry_get_value (entry)));
+        }
+    }
 
   return g_steal_pointer (&content_rating);
```

Comment:
Looks like we're returning an empty hash table in case the info is missing, whereas previously we returned NULL in that case. Not sure if that's going to cause problems.